ui/src/app/base/components/TableActionsDropdown/TableActionsDropdown.tsx

```diff
@@ -0,0 +1,48 @@
+import type { Props as ButtonProps } from "@canonical/react-components/dist/components/Button";
+
+import TableMenu from "app/base/components/TableMenu";
+
+export type TableAction<A> = {
+  label: string;
+  show?: boolean;
+  type: A;
+};
+
+// This allows the "data-test" attribute to be used for the action links, which
+// is technically not a valid HTML prop, but we use it throughout maas-ui.
+type TableActionsLink = ButtonProps & { "data-test": string };
+
+type Props<A> = {
+  actions: TableAction<A>[];
+  disabled?: boolean;
+  onActionClick: (action: A) => void;
+};
+
+const TableActionsDropdown = <A extends string>({
+  actions,
+  disabled = false,
+  onActionClick,
+}: Props<A>): JSX.Element => {
+  const actionLinks = actions.reduce<TableActionsLink[]>((links, action) => {
+    if (!(action.show === false)) {
+      // Show actions that do not explicitly set show to false.
```

Comment:
Why does it need to be an explicit `false`?

Reply:
Just so if the action object doesn't have a `show` key, the assumption is to show the action anyway. I could have rewritten the  object to have a `hide` key, then `if (!action.hide)` would do the same thing but I think `show` is clearer in the list of actions.